File: cpp/countfrequency.cpp
#include<bits/stdc++.h>
using namespace std;
void frequency(vector<int>a,int n)
{
    vector<bool>visited(n,false);
    for(int i=0;i<n;i++)
    {
        if(visited[i]==true)
        continue;
        
            int count=1;
            for(int j=i+1;j<n;j++)
            {
                if(a[i]==a[j])
                {
                    count++;
                    visited[j]=true;
                }
            }
        cout<<a[i]<<" "<<count<<endl;
           
    }
}
int main()
{
    int n;
    cin>>n;
    vector<int>a;
    for(int i=0;i<n;i++)
    {
        int x;
        cin>>x;
        a.push_back(x);
    }
    frequency(a,n);
    return 0;
}

//TC=O(n*n) SC=O(n)

// void frequency(vector<int>a,int n)
// {
//     unordered_map<int,int>mp;
//     for(int i=0;i<n;i++)
//     {
//         mp[a[i]]++;
//     }

//     for(auto it : mp)
//     {
//         cout<<it.first<<" "<<it.second;
//     }
// }

// TC=O(n) SC=O(n)

File: python/countfrequency.py
def fre(a,n):
    visited = [False]*n
    for i in range(n):
        if visited[i]:
            continue
        count = 1
        for j in range(i+1,n):
            if a[i]==a[j]:
                visited[j]=True
                count+=1
        print(a[i],"=",count)


if __name__ == "__main__":
    n=int(input())
    a = list(map(int,input().split()))
    fre(a,n)
    
    
    
# def Frequency(arr, n):
#     mp = {}
#     for i in range(n):
#         if arr[i] in mp:
#             mp[arr[i]] += 1
#         else:
#             mp[arr[i]] = 1
#     for x in mp:
#         print(x, mp[x])